### PR TITLE
Lt 4490 cache configuration and proxy

### PR DESCRIPTION
**Pull request by tarurar**

### CODE REVIEW

File: src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs [hunk 1]
```diff
@@ -9,9 +9,9 @@ namespace Lykke.Snow.Notifications.Domain.Model
     /// <summary>
     /// Device configuration
     /// </summary>
-    public sealed class DeviceConfiguration
+    public sealed class DeviceConfiguration : IEquatable<DeviceConfiguration>
     {
-        public class Notification
+        public sealed class Notification : IEquatable<Notification>
```
Comment (fatihdumanli) [inline src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs:14]: Why is this needed? Do we compare any configuration with another?

File: src/Lykke.Snow.Notifications.Domain/Model/DeviceRegistration.cs [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 namespace Lykke.Snow.Notifications.Domain.Model
 {
-    public sealed class DeviceRegistration
+    public sealed class DeviceRegistration : IEquatable<DeviceRegistration>
```
Comment (fatihdumanli) [inline src/Lykke.Snow.Notifications.Domain/Model/DeviceRegistration.cs:5]: Same

File: src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs (hunk 1 above)
Comment (tarurar) [reply src/Lykke.Snow.Notifications.Domain/Model/DeviceConfiguration.cs:14]: We have a test which checks that transformation domain model -> entity and back gives the origin model: https://github.com/LykkeBusiness/Lykke.Snow.Notifications/pull/11/files#diff-93d1e72291fbac785080716410c88fcc9321128c1b9a91c88e1ccc2cae3c360cR45. 
The test handles mapping issues when properties are lost.

File: README.md [hunk 3]
```diff
@@ -77,6 +77,31 @@ And the languages should be put in lowercase format. (i.e. `en`, `es`, `de`)
 
 All notification types can be found at [here](./src/Lykke.Snow.Notifications.Domain/Enums/NotificationType.cs). 
 
+### 4. Proxy
+
+Proxy is optional. Being set, it will be used for all FCM outgoing requests.
+
+```json
+{
```
[the remark this answers is not in the document]
Comment (tarurar) [reply README.md:85]: Down

**Review by fatihdumanli — APPROVED** [review]
